### PR TITLE
Wildcard V2: <Icon /> Implementation

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
### Wildcard implementation of the <Icon /> component.

See [Wildcard V2 - Planned work](https://docs.google.com/document/d/1NisbJPiadtt5jQw4vUUYJOr8dD6Urwn5V0mVq2wt6bE/edit#heading=h.g4cw92w3ouhw) for more context
## Acceptance criteria

- [x] Component is implemented within Wildcard
- [x] Component has tests
- [x] Component is accessible
- [ ] Component matches [Figma designs](https://www.figma.com/file/NIsN34NH7lPu04olBzddTw/Wildcard-Design-System?node-id=908%3A1)

## Refs
* [Gitstart Task](https://app.gitstart.com/clients/sourcegraph/tickets/SG-27667) 
* [SourceGraph Issue](https://github.com/sourcegraph/sourcegraph/issues/27667)

## Implementation details

1.  Add new <Icon /> to the Wildcard package.
    - A wrapper component that receives svg and applies the icon-inline class.
    - Copy [the existing global styles](https://github.com/sourcegraph/sourcegraph/blob/main/client/shared/src/global-styles/icons.scss) to the component CSS module.
    - Remove redundant classes that we don't use in the codebase.
2.  Create a basic Storybook story that shows component usage.
3.  Use [the <Button /> component](https://github.com/sourcegraph/sourcegraph/tree/main/client/wildcard/src/components/Button) as a blueprint for the new component.
4.  Component API:

`<Icon svg={CloudOutlineIcon} />`

### CODE REVIEW

File: client/wildcard/src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import classNames from 'classnames'
+import React from 'react'
+
+import { ICON_SIZES } from './constants'
+import styles from './Icon.module.scss'
+interface IconProps {
```
Comment (valerybugakov) [inline client/wildcard/src/components/Icon/Icon.tsx:6]:
```suggestion
import styles from './Icon.module.scss'

interface IconProps {
```

File: client/wildcard/src/components/Icon/Icon.story.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            type: 'figma',
+            name: 'Figma',
+            url: 'https://www.figma.com/file/NIsN34NH7lPu04olBzddTw/Wildcard-Design-System?node-id=908%3A1',
+        },
+    },
+}
+
+export default Story
+
+export const Simple = () => (
```
Comment (valerybugakov) [inline client/wildcard/src/components/Icon/Icon.story.tsx:32]:
```suggestion
import { Story } from '@storybook/react'

export const Simple: Story = () => (
```

File: client/wildcard/src/components/Icon/Icon.story.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            <BrandedStory styles={webStyles}>{() => <div className="container mt-3">{story()}</div>}</BrandedStory>
+        ),
+    ],
+
+    parameters: {
+        component: Icon,
+        design: {
+            type: 'figma',
+            name: 'Figma',
+            url: 'https://www.figma.com/file/NIsN34NH7lPu04olBzddTw/Wildcard-Design-System?node-id=908%3A1',
```
Comment (valerybugakov) [inline client/wildcard/src/components/Icon/Icon.story.tsx:25]: Does it point to a specific Icon frame in Figma? It opens the frame with Form Elements for me.
If we don't have a relevant Figma frame, let's delete this configuration parameter.

**Review by valerybugakov — CHANGES_REQUESTED** [review]

**Comment by gitstart-sourcegraph** [comment]
Hi @valerybugakov requested fixes has been added. Thanks

**Review by valerybugakov — APPROVED** [review]
✅

**Review by valerybugakov — APPROVED** [review]
Oh, I see that there's a failed CI check. If it's infrastructure issue, let's restart the job to ensure that it's green before merging.

**Pull request merged**